include/kllvm/util/temporary_file.h

```diff
@@ -38,6 +38,11 @@ class temporary_file {
     if (!temp_c_file.get()) {
       temp_c_file = temp_c_file_type(fdopen(temp_fd, mode.data()), &fclose);
     }
+
+    if (!temp_c_file.get()) {
+      std::runtime_error("Could not open file " + temp_filename);
+    }
+
```

Comment:
I added this second check to sanitize the `fdopen` function and to include the [`check_fopen`](https://github.com/runtimeverification/llvm-backend/blob/aa0a2278a8da15e298a0aa616928ac98e2fd006a/tools/kore-strip/main.cpp#L40) behavior from `kore-strip`, any thoughts on this @Baltoli?